### PR TITLE
⚡️ Implement business logic for ConditionalAvailabilityProductPageSearch

### PR DESCRIPTION
**Pull request by gengjozsef**
- Implement Expanders, Reader, BusinessFactory, etc
- Implement Unit Test for the Implemented Expanders, Reader, BusinessFactory, etc

### CODE REVIEW

File: bundles/conditional-availability-product-page-search/src/FondOfImpala/Shared/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchConfig.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+namespace FondOfImpala\Shared\ConditionalAvailabilityProductPageSearch;
+
+use Spryker\Shared\Kernel\AbstractBundleConfig;
+
+class ConditionalAvailabilityProductPageSearchConfig extends AbstractBundleConfig
+{
+    /**
+     * @var string
+     */
+    public const PLUGIN_STOCK_STATUS_DATA = 'PLUGIN_STOCK_STATUS_DATA';
```
Comment (daniel-rose) [inline bundles/conditional-availability-product-page-search/src/FondOfImpala/Shared/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchConfig.php:12]: Not correct position for this entry. If you want to use it, please move it to dependency provider.

File: bundles/conditional-availability-product-page-search/src/FondOfImpala/Zed/ConditionalAvailabilityProductPageSearch/Business/ProductPage/ProductPageDataExpander.php [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        $conditionalAvailabilityPeriodTransfer = array_shift($conditionalAvailabilityPeriods);
+
+        if ($conditionalAvailabilityPeriodTransfer->getQuantity() > 0) {
+            return ConditionalAvailabilityProductPageSearchConfig::STOCK_STATUS_IN_STOCK;
+        }
+
+        foreach ($conditionalAvailabilityPeriods as $conditionalAvailabilityPeriodTransfer) {
+            /** @var \Generated\Shared\Transfer\ConditionalAvailabilityPeriodTransfer $conditionalAvailabilityPeriodTransfer */
+            if ($conditionalAvailabilityPeriodTransfer->getQuantity() > 0) {
+                return ConditionalAvailabilityProductPageSearchConfig::STOCK_STATUS_OUT_OF_STOCK;
```
Comment (daniel-rose) [inline bundles/conditional-availability-product-page-search/src/FondOfImpala/Zed/ConditionalAvailabilityProductPageSearch/Business/ProductPage/ProductPageDataExpander.php:150]: Are you sure?

**Review by daniel-rose — APPROVED** [review]
Please use typed properties in all classes...

File: bundles/conditional-availability-product-page-search/src/FondOfImpala/Shared/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchConfig.php (hunk 1 above)
Comment (gengjozsef) [reply bundles/conditional-availability-product-page-search/src/FondOfImpala/Shared/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchConfig.php:12]: @daniel-rose as we have agreed i will leave the const.

File: bundles/conditional-availability-product-page-search/src/FondOfImpala/Zed/ConditionalAvailabilityProductPageSearch/Business/ProductPage/ProductPageDataExpander.php (hunk 2 above)
Comment (gengjozsef) [reply bundles/conditional-availability-product-page-search/src/FondOfImpala/Zed/ConditionalAvailabilityProductPageSearch/Business/ProductPage/ProductPageDataExpander.php]: @daniel-rose thank you for the remark you are correct the `ConditionalAvailabilityProductPageSearchConfig::STOCK_STATUS_OUT_OF_STOCK;` is wrong here. I've corrected it

**Pull request merged**